### PR TITLE
Radiative Rates Test

### PR DESCRIPTION
**Pull request by RishmitaR**
### :pencil: Description

**Type:**  :vertical_traffic_light: `testing` 

Updated Radiative Rates pytests from original gsoc on boarding task https://github.com/tardis-sn/tardis/pull/3529. This file adds tests cases to ensure that asserts are properly triggered. 

### :pushpin: Resources

PR for updated regression data: https://github.com/tardis-sn/tardis-regression-data/pull/96

### :vertical_traffic_light: Testing

How did you test these changes?

- [ ] Testing pipeline
- [ ] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [ ] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

File: .zenodo.json
Comment (connor-mcclellan) [inline .zenodo.json:1]: Why is this modified in so many places? Check that you didn't accidentally use an automatic formatter on this file.

File: CITATION.cff
Comment (connor-mcclellan) [inline CITATION.cff:1]: Again, hard to separate actual edits here from shuffling the names around. We don't want to include all this in the commit.

File: README.rst
Comment (connor-mcclellan) [inline README.rst:1]: Also undo these changes

File: pytest
Comment (connor-mcclellan) [inline pytest:1]: I'm concerned about this too

File: .orcid.csv [hunk 1]
```diff
@@ -30,7 +30,10 @@ mohant11@msu.edu,0000-0003-1197-6693
 b.connor.mcc@gmail.com,0000-0002-6040-8281
 aaryandadu5157@gmail.com,0009-0005-3670-8777
 shreyanssaraf10@gmail.com,0009-0006-0201-8249
+abhayraj916146@gmail.com,0009-0003-8477-2963
 araut7798@gmail.com ,0009-0009-4508-2218
+rishmita.rao12@gmail.com,0009-0003-3964-2107
+
```
Comment (connor-mcclellan) [inline .orcid.csv]: Unnecessary whitespace, duplicate entry

File: .mailmap [hunk 2]
```diff
@@ -317,6 +317,7 @@ Riddhi Gangbhoj <riddhigangbhoj76@gmail.com> riddhigangbhoj <riddhigangbhoj76@gm
 
 Connor McClellan <b.connor.mcc@gmail.com>
 Aditya Raut <araut7798@gmail.com>
-Rishmita Rao <rishmita.rao12@gmail.com> 
+
+Rishmita Rao <rishmita.rao12@gmail.com>
```
Comment (connor-mcclellan) [inline .mailmap]: Unnecessary whitespace

File: tardis/plasma/tests/data/plasma_base_test_config.yml
Comment (connor-mcclellan) [inline tardis/plasma/tests/data/plasma_base_test_config.yml:1]: What's the justification for changing this?

File: tardis/plasma/equilibrium/tests/test_radiative_rates.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    scope="function",
+    params=["invalid_index", "invalid_column", "invalid_lower_higher"]
+)
+def invalid_coefficients(request, regression_data):
+    invalid_index_file = (
+        regression_data.regression_data_path
+        / "testdata"
+        / "plasma_tests"
+        / "radiative_rates_test_xfails.h5"
+    )
```
Comment (connor-mcclellan) [inline tardis/plasma/equilibrium/tests/test_radiative_rates.py]: Could this be done by constructing an invalid coefficients dataframe here, rather than requiring it as a regression data file? In some other simple test cases (particularly edge cases) we just build an array inline with the test, so if that can be done here as well I think that'd be preferred. It wouldn't be too hard to assemble a dataframe with the right shape that's full of zeros, for example

File: docs/resources/credits.rst
Comment (connor-mcclellan) [inline docs/resources/credits.rst:1]: These don't seem like necessary changes?

**Review by connor-mcclellan — DISMISSED** [review]

File: tardis/transport/montecarlo/interaction_event_callers.py [hunk 4]
```diff
@@ -4,6 +4,8 @@
 import tardis.transport.montecarlo.configuration.montecarlo_globals as montecarlo_globals
 from tardis.opacities.opacity_state_numba import OpacityStateNumba
 from tardis.opacities.opacity_state_numba_iip import OpacityStateNumbaIIP
+from tardis.opacities.opacity_state_numba import OpacityStateNumba
```
Comment (jvshields) [inline tardis/transport/montecarlo/interaction_event_callers.py]: Is this just a double import? It seems like these are imported directly above.

File: tardis/plasma/equilibrium/tests/test_radiative_rates.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    invalid_df = pd.read_hdf(invalid_index_file, request.param)
```
Comment (connor-mcclellan) [inline tardis/plasma/equilibrium/tests/test_radiative_rates.py]: sync dataframe, not read hdf

File: tardis/plasma/equilibrium/tests/test_radiative_rates.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+import pytest
+
+from tardis.io.atom_data import AtomData
+from tardis.plasma.equilibrium.rates.radiative_rates import RadiativeRatesSolver
+from tardis.plasma.radiation_field.planck_rad_field import (
+    PlanckianRadiationField,
+)
+
+
+@pytest.fixture(scope="class")
```
Comment (RishmitaR) [inline tardis/plasma/equilibrium/tests/test_radiative_rates.py]: 1. no need for a fixture for einstein_coeffs
2. the way I'm slicing the radiative_transitions is weird, I should fix it and mark it as an issue where the other part of the code is doing the same thing

File: tardis/plasma/equilibrium/tests/conftest.py [hunk 7]
```diff
@@ -36,6 +36,15 @@ def new_chianti_atomic_dataset_si(tardis_regression_path):
     )
     return AtomData.from_hdf(atomic_data_fname)
 
+@pytest.fixture
+def new_chianti_atomic_dataset(tardis_regression_path):
```
Comment (andrewfullard) [inline tardis/plasma/equilibrium/tests/conftest.py:40]: This fixture already exists in the atom_data.py fixtures file. I realise it's one I created back when I was working on the rates so feel free to remove it and replace.

File: tardis/plasma/equilibrium/tests/test_radiative_rates.py [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+    einstein_coefficients_df = new_chianti_atomic_dataset.lines.xs((1,0),drop_level=False)
+    solver = RadiativeRatesSolver(einstein_coefficients_df)
+    actual_radiative_rates = solver.solve(mock_radiation_field)
+    expected_radiative_rates = regression_data.sync_dataframe(
+        actual_radiative_rates, key="solved_radiative_rates"
+    )
+    pdt.assert_frame_equal(actual_radiative_rates,expected_radiative_rates,atol=0,rtol=1e-15)
+
+@pytest.mark.xfail(strict=True, raises=AssertionError)
+def test_invalid_coefficients(invalid_coefficients):
```
Comment (andrewfullard) [inline tardis/plasma/equilibrium/tests/test_radiative_rates.py:103]: I don't think this is a very useful test

**Review by andrewfullard — CHANGES_REQUESTED** [review]

File: tardis/plasma/equilibrium/tests/test_radiative_rates.py (hunk 8 above)
Comment (RishmitaR) [reply tardis/plasma/equilibrium/tests/test_radiative_rates.py:103]: Is it specifically because of the data frames being used to check it or in general is it not a useful test because it's just a check in initialization?

Comment (andrewfullard) [reply tardis/plasma/equilibrium/tests/test_radiative_rates.py:103]: Because it simply checks that an assertion is raised with invalid input. If the function is not originally designed to do that, it's not worth testing.

**Review by andrewfullard — DISMISSED** [review]
Approved pending tests passing with merged regression data

File: tardis/plasma/equilibrium/tests/conftest.py [hunk 9]
```diff
@@ -56,9 +56,9 @@ def hydrogen_atomic_data_fname(tardis_regression_path):
     return atomic_data_fname
 
 
-@pytest.fixture(params=[(14, 1, slice(None), slice(None))])
-def radiative_transitions(new_chianti_atomic_dataset_si, request):
-    return new_chianti_atomic_dataset_si.lines.loc[request.param, :]
+@pytest.fixture
+def radiative_rate_solver(radiative_transitions):
```
Comment (andrewfullard) [inline tardis/plasma/equilibrium/tests/conftest.py:73]: It looks like you unintentionally removed radiative_transitions, which is causing some cascading failures

**Review by andrewfullard — APPROVED** [review]

File: tardis/plasma/equilibrium/tests/test_radiative_rates.py [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+    temperature = [10000, 20000]* u.K
+    return PlanckianRadiationField(temperature=temperature)
+
+def test_radiative_rate_solver_init(new_chianti_atomic_dataset,regression_data):
+    einstein_coefficients_df = new_chianti_atomic_dataset.lines.xs((1,0),drop_level=False)
+    solver = RadiativeRatesSolver(einstein_coefficients_df)
+    actual_einstein_coeffs = solver.einstein_coefficients
+    expected_einstein_coeffs = regression_data.sync_dataframe(
+        actual_einstein_coeffs, key="einstein_coeffs")
+    # CAN DO A NORMAL ASSERT WHEN ATOMIC DATA IS UPDATED FOR PANDAS 3.X
```
Comment (jvshields) [inline tardis/plasma/equilibrium/tests/test_radiative_rates.py:87]: This comment can be removed now right?

**Review by jvshields — APPROVED** [review]
